### PR TITLE
Deep link into the app after mobile login

### PR DESCRIPTION
**Pull request by leogrim**
@DerekBurch @andrewconner is that it?

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 1]
```diff
@@ -293,7 +295,8 @@ class AuthController @Inject() (
           userRequest.session.get(SecureSocial.OriginalUrlKey) map { url =>
             Redirect(url).withSession(userRequest.session - SecureSocial.OriginalUrlKey)
           } getOrElse {
-            Redirect("/")
+            if (req.agent.isMobile) Ok(views.html.mobile.deepLinkRedirect(deepLinkRouter.generateRedirect(DeepLinkRouter.viewHomepage).get, DeepLinkRouter.viewHomepage))
+            else Redirect("/")
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:299]: @DerekBurch will the login process happen in our the user's mobile browser or in our app's browser? If the latter, I can be more restrictive here

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:299]: @andrewconner alternatively, is there any way for the app to set `SecureSocial.OriginalUrlKey`?

**Comment by andrewconner** [comment]
Won't this force the app to open or the user to install the app? Are we not supporting mobile web anymore?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:299]: Background? What's the use case / problem?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:299]: Slack login on mobile

**Comment by leogrim** [comment]
Just trying to support Slack login on mobile, via web

**Comment by andrewconner** [comment]
I see. Mind walking me through the intended flow? If this is intended to happen in their browser, cookies won't transfer (the user won't be logged in). If it's intended to happen in the mobile client (and Derek supported this in a separate web view), you won't need a deep link router anyways (they're in the app already).

**Comment by leogrim** [comment]
I'm not sure whether the flow is intended to happen in their browser or the app (hence my question to Derek). Taking a step back, trying to support Slack auth on mobile via mobile web - as opposed to implementing a full OAuth back and forth in the app, which we could do but that's coming a bit late for the next releases.

**Comment by andrewconner** [comment]
I see. Then you'd probably want a web view for this purpose, go through web login, and then he could grab the cookies of the web view (you can do that, right?). He should be able to just detect that it's done when it redirects to `/` or whatever. @DerekBurch thoughts?

Sorry I'm a bit behind here.

**Comment by andrewconner** [comment]
In other words, if my suggestion is good, there's not really anything needed on backend

**Comment by DerekBurch** [comment]
I considered the approach of reading the cookies from a webview and while that approach is free from backend implementation it is most likely more complicated on the front end. I would probably need to make new webview activities/viewcontrollers for each platform and try to detect when you are redirected home (And ideally not show the page at all. Which probably is possible, but I have not tried.). I figured the intent was "simpler" because then each app only really needed to listen for the login intent (And initially redirect the user to chrome/safari for the login flow [no webview required]). Though to be fair the webview implementation may appear as "nicer" for the end user. Also I am not super crazy about listening for "/" redirection in case we ever redirect the user somewhere else after logging in (Like perhaps to their team page?).

I guess we just need to decide what we think is the "best" choice overall. Either is mostly fine with me.

**Comment by andrewconner** [comment]
Alright, thanks for the background. How do you suggest we get the session into the app?

**Pull request closed without merging**